pkg/resources/grant_privileges_to_share.go

```diff
@@ -0,0 +1,474 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"slices"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var grantPrivilegesToShareGrantExactlyOneOfValidation = []string{
+	"database_name",
+	"schema_name",
+	// "function_name",
+	"table_name",
+	"all_tables_in_schema",
+	"tag_name",
+	"view_name",
+}
+
+var grantPrivilegesToShareSchema = map[string]*schema.Schema{
+	"share_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "The fully qualified name of the share on which privileges will be granted.",
+	},
+	"privileges": {
+		Type:        schema.TypeSet,
+		Required:    true,
+		Description: "The privileges to grant on the share.",
```

Comment:
the privileges list changes depending on the type of the resource. Can we at least direct customer to https://docs.snowflake.com/en/sql-reference/sql/grant-privilege-share#syntax for the list of allowed privileges?

Reply:
added